contracts/HubPool.sol

```diff
@@ -139,11 +202,34 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
 
     function liquidityUtilizationPostRelay(address token, uint256 relayedAmount) public returns (uint256) {}
 
+    /*************************************************
+     *             DATA WORKER FUNCTIONS             *
+     *************************************************/
+
     function initiateRelayerRefund(
-        uint256[] memory bundleEvaluationBlockNumberForChain,
-        bytes32 chainBatchRepaymentProof,
-        bytes32 relayerRepaymentDistributionProof
-    ) public {}
+        uint256[] memory bundleEvaluationBlockNumbers,
+        bytes32 poolRebalanceProof,
+        bytes32 destinationDistributionProof
+    ) public {
+        relayerRefundRequests.push(
+            RelayerRefundRequest({
+                poolRebalanceProof: poolRebalanceProof,
+                destinationDistributionProof: destinationDistributionProof,
+                status: RefundRequestStatus.Pending
+            })
+        );
+
+        // Pull bonds from from the caller.
+        bondToken.safeTransferFrom(msg.sender, address(this), bondAmount);
```

Comment:
This is annoying, but how do we deal with a changing bond amount mid-refund? I know this code is to pull the bond, but do you think we need to store the bond amount for each refund request?

Reply:
just dont change it if there is a pending refund I guess.